Remove Notebook and mantidpython

DanielMurphy22 opened the pull request:
Separately the [PythonNotebook docs page](https://docs.mantidproject.org/v6.1.0/concepts/PythonNotebook.html) has been updated on main (not on the website yet) [in this PR](https://github.com/mantidproject/mantid/pull/34450).

**Description of work.**
- Remove the mantidpython and mantidnotebook. They were both separate launcher scripts, and the mantidpython was also used for running our systemtests and python unittests. Now, to access the mantid api, the PYTHONPATH is set to the build/bin directory to run the tests. With an installed package of mantidworkbench, you can just launch the bundled python as detailed here: https://github.com/mantidproject/mantid/pull/34450/files#diff-dab7a662c04b0c0ab7ee1ec2aa593b9d4a8830b2a2659f08ee43e550047adf40R26-R28 Users are expected to install the notebook python module themselves.
- Removed the .deb and .rpm package scripts
- Removed all references to ``mantidpython`` in code and docs (apart from old release notes)
- Removed all references to ``notebook`` in code (apart from algorithms that write notebooks such as NotebookBuilder, NotebookWriter, GenerateIPythonNotebook, Generate Notebook, DataProcessorWidget)
and docs (apart from old release notes and a few parts of the dev-docs that I think are fine!) 
**To test:**
1) Jenkins
- PR builds check that unit tests ran
- Run SEPARATE build_package_with_branch jobs (must be separate for systemtests to run) for Windows and MacOS to check the systemtests run ( I got a recent version of this PR to pass on build numbers 78 and 76 respectively).

2) Check Jemalloc is used for tests on Linux
- On Linux, run the test-pr command to get this branch, activate the dev conda environment and build
- Check jemalloc is used for unittests with this command giving jemalloc in it's output: ``ctest -V -R AnalysisData -N``
- start the system tests running with `.systemtest` in the build folder. Wait til they are actually running and not downloading data. 
- Then in another terminal run `ps ax | grep runSystemTests` and note the process ID
- Now run `cat /proc/PID/smaps | grep jemalloc` but replace PID with the process ID from the last step and you should get some output which confirms jemalloc is being used by the systemtests!

3) check the docs updates are okay (a code review should suffice).
4) Finally some git grepping!
- In the source directory run `git grep -n "mantidpython"` and check the list is as described in the "Description of work" above.
- Run `git grep -n "notebook"` and check the list is as described in the "Description of work" above. You may wish to cd into docs or dev-docs to make the list more digest-able.

Fixes #34388

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

DanielMurphy22 committed:
Remove mantidpython and add pythonpath for tests

remove rpm and deb scripts

DanielMurphy22 committed:
Remove mantid_notebook

DanielMurphy22 commented:
should consider how to preserve the custom memory allocation from jemalloc on Linux, currently part of mantidpython

DanielMurphy22 committed:
Merge branch 'main' into remove_notebook_34388

DanielMurphy22 committed:
Merge remote-tracking branch 'origin/main' into remove_notebook_34388

DanielMurphy22 committed:
Remove all code file references to mantidpython and notebook

Altered notebook docs references
TODO review mantidpython references in docs

DanielMurphy22 committed:
Set PYTHONPATH for Windows Systemtests

DanielMurphy22 committed:
Use Jemalloc for Unit and System tests

DanielMurphy22 committed:
Remove mantidpython from LinuxPaackageScripts after merge

Also remove mantidpython references in docs

DanielMurphy22 committed:
Add release note

robertapplin reviewed (changes requested):
Nice clean changes! I have a few suggestions and questions about the removal of rpm/deb files

DanielMurphy22 committed:
Merge branch 'main' into remove_notebook_34388

DanielMurphy22 committed:
Requested Changes

Rework jemalloc if statements in PyUnitTest
reinstate AddPythonPath
Reword mantidpython in Core test instructions

martyngigg commented on buildconfig/CMake/Packaging/rpm/scripts/rpm_post_install.sh.in:
```diff
@@ -1,44 +0,0 @@
-#!/bin/sh
```
It is now yes as we no longer support RHEL 7

martyngigg commented on Testing/SystemTests/scripts/mantidinstaller.py:
```diff
@@ -166,156 +136,9 @@ def do_uninstall(self):
         raise NotImplementedError("Override the do_uninstall method")
 
 
-class NSISInstaller(MantidInstaller):
```
Yep this is fine too as they are not being used.

martyngigg commented on buildconfig/CMake/Packaging/AddPythonPath.py.in:
```diff
@@ -78,4 +78,4 @@ with open(pathfile, 'w') as f:
         if not directory.endswith(os.sep):
```
Is this file used anywhere? I notice it as a lot of Python 2.7 code in it too.

robertapplin commented:
All tests complete and code is looking much cleaner, so nice work! I just need to build the system tests on Windows and OSX using the build_packages_from_branch pipeline, once the release is out the door

DanielMurphy22 committed:
Merge remote-tracking branch 'origin/main' into remove_notebook_34388

DanielMurphy22 commented:
> I just need to build the system tests on Windows and OSX using the build_packages_from_branch pipeline, once the release is out the door

After build package from branch 100 and 101 failed due to a problem on main which is now fixed, I've merged with current main and re-run the build with package jobs as 113 and 114 https://builds.mantidproject.org/job/build_packages_from_branch/

robertapplin reviewed (approved):
Good work with this! The vast majority of references to mantidpython are now removed, and some unneeded scripts have also been removed.

The system tests are passing on windows, OSX and linux :+1:
